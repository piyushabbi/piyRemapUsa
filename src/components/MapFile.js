import React from 'react';
import PropTypes from 'prop-types';
import data from '../data/states_info.json';
import USAState from './USAState';

class USAMap extends React.Component {
	state = {
		showCode: false
	};

	clickHandler = stateAbbreviation => {
		this.props.onClick(stateAbbreviation);
	};

	fillStateColor = state => {
		if (
			this.props.customize &&
			this.props.customize[state] &&
			this.props.customize[state].fill
		) {
			return this.props.customize[state].fill;
		}

		return this.props.defaultFill;
	};

	stateClickHandler = state => {
		if (
			this.props.customize &&
			this.props.customize[state] &&
			this.props.customize[state].clickHandler
		) {
			return this.props.customize[state].clickHandler;
		}
		return this.clickHandler;
	};

	mouseEventHandler = (stateAbbr, stateFull) => {
		//alert(state);
		this.props.titleHandler(stateAbbr, stateFull);
	};

	buildPaths = () => {
		let paths = [];
		for (let stateKey in data) {
			const path = (
				<USAState
					key={stateKey}
					dimensions={data[stateKey]['dimensions']}
					state={stateKey}
					fill={this.fillStateColor(stateKey)}
					onClickState={this.stateClickHandler(stateKey)}
					mouseEventHandler={this.mouseEventHandler}
					name={data[stateKey]['name']}
				/>
			);
			paths.push(path);
		}
		return paths;
	};

	render() {
		return (
			<React.Fragment>
				<svg
					className="us-state-map"
					xmlns="http://www.w3.org/2000/svg"
					width={this.props.width}
					height={this.props.height}
					viewBox="0 0 959 593"
				>
					<title>{this.props.title}</title>
					<g className="outlines">
						{this.buildPaths()}
						<g className="DC state">
							<path
								className="DC1"
								fill={this.fillStateColor('DC1')}
								d="M801.8,253.8 l-1.1-1.6 -1-0.8 1.1-1.6 2.2,1.5z"
							/>
							<circle
								className="DC2"
								onClick={this.clickHandler}
								data-name={'DC'}
								fill={this.fillStateColor('DC2')}
								stroke="#FFFFFF"
								strokeWidth="1.5"
								cx="801.3"
								cy="251.8"
								r="5"
								opacity="1"
							/>
						</g>
					</g>
				</svg>

				<div className="show-json">
					<button
						onClick={() => {
							this.setState(prevState => ({
								showCode: !prevState.showCode
							}));
						}}
					>
						Toggle States Data
					</button>
					{/* Total States JSON*/}
					{this.state.showCode && (
						<pre>
							<code>{JSON.stringify(data, undefined, 4)}</code>
						</pre>
 					)}
				</div>
			</React.Fragment>
		);
	}
}

USAMap.propTypes = {
	onClick: PropTypes.func,
	width: PropTypes.number,
	height: PropTypes.number,
	title: PropTypes.string,
	defaultFill: PropTypes.string,
	customize: PropTypes.object
};

USAMap.defaultProps = {
	onClick: () => {},
	width: 959,
	height: 593,
	defaultFill: '#D3D3D3',
	title: 'Yo Man! Wadd up......',
	customize: {}
};

export default USAMap;
