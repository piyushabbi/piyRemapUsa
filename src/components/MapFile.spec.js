import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import MapFile from './MapFile';

describe('MapFile Component', () => {
	let wrapper;

	beforeEach(() => {
		const titleHandler = jest.fn();
		wrapper = shallow(<MapFile />);
	});

	it('renders snapshot correctly', () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('executes mouse event', () => {
		// console.log(wrapper.instance())
	});
});
