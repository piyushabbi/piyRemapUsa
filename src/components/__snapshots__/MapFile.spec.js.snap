// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapFile Component renders snapshot correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <USAMap
    customize={Object {}}
    defaultFill="#D3D3D3"
    height={593}
    onClick={[Function]}
    title="Yo Man! Wadd up......"
    width={959}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <svg
        className="us-state-map"
        height={593}
        viewBox="0 0 959 593"
        width={959}
        xmlns="http://www.w3.org/2000/svg"
>
        <title>
                Yo Man! Wadd up......
        </title>
        <g
                className="outlines"
        >
                <g
                        className="DC state"
                >
                        <path
                                className="DC1"
                                d="M801.8,253.8 l-1.1-1.6 -1-0.8 1.1-1.6 2.2,1.5z"
                                fill="#D3D3D3"
                        />
                        <circle
                                className="DC2"
                                cx="801.3"
                                cy="251.8"
                                data-name="DC"
                                fill="#D3D3D3"
                                onClick={undefined}
                                opacity="1"
                                r="5"
                                stroke="#FFFFFF"
                                strokeWidth="1.5"
                        />
                </g>
        </g>
</svg>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <title>
            Yo Man! Wadd up......
</title>,
          <g
            className="outlines"
>
            <g
                        className="DC state"
            >
                        <path
                                    className="DC1"
                                    d="M801.8,253.8 l-1.1-1.6 -1-0.8 1.1-1.6 2.2,1.5z"
                                    fill="#D3D3D3"
                        />
                        <circle
                                    className="DC2"
                                    cx="801.3"
                                    cy="251.8"
                                    data-name="DC"
                                    fill="#D3D3D3"
                                    onClick={undefined}
                                    opacity="1"
                                    r="5"
                                    stroke="#FFFFFF"
                                    strokeWidth="1.5"
                        />
            </g>
</g>,
        ],
        "className": "us-state-map",
        "height": 593,
        "viewBox": "0 0 959 593",
        "width": 959,
        "xmlns": "http://www.w3.org/2000/svg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Yo Man! Wadd up......",
          },
          "ref": null,
          "rendered": "Yo Man! Wadd up......",
          "type": "title",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [],
              <g
                className="DC state"
>
                <path
                                className="DC1"
                                d="M801.8,253.8 l-1.1-1.6 -1-0.8 1.1-1.6 2.2,1.5z"
                                fill="#D3D3D3"
                />
                <circle
                                className="DC2"
                                cx="801.3"
                                cy="251.8"
                                data-name="DC"
                                fill="#D3D3D3"
                                onClick={undefined}
                                opacity="1"
                                r="5"
                                stroke="#FFFFFF"
                                strokeWidth="1.5"
                />
</g>,
            ],
            "className": "outlines",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <path
                    className="DC1"
                    d="M801.8,253.8 l-1.1-1.6 -1-0.8 1.1-1.6 2.2,1.5z"
                    fill="#D3D3D3"
/>,
                  <circle
                    className="DC2"
                    cx="801.3"
                    cy="251.8"
                    data-name="DC"
                    fill="#D3D3D3"
                    onClick={undefined}
                    opacity="1"
                    r="5"
                    stroke="#FFFFFF"
                    strokeWidth="1.5"
/>,
                ],
                "className": "DC state",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "DC1",
                    "d": "M801.8,253.8 l-1.1-1.6 -1-0.8 1.1-1.6 2.2,1.5z",
                    "fill": "#D3D3D3",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "path",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "DC2",
                    "cx": "801.3",
                    "cy": "251.8",
                    "data-name": "DC",
                    "fill": "#D3D3D3",
                    "onClick": undefined,
                    "opacity": "1",
                    "r": "5",
                    "stroke": "#FFFFFF",
                    "strokeWidth": "1.5",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "circle",
                },
              ],
              "type": "g",
            },
          ],
          "type": "g",
        },
      ],
      "type": "svg",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <svg
          className="us-state-map"
          height={593}
          viewBox="0 0 959 593"
          width={959}
          xmlns="http://www.w3.org/2000/svg"
>
          <title>
                    Yo Man! Wadd up......
          </title>
          <g
                    className="outlines"
          >
                    <g
                              className="DC state"
                    >
                              <path
                                        className="DC1"
                                        d="M801.8,253.8 l-1.1-1.6 -1-0.8 1.1-1.6 2.2,1.5z"
                                        fill="#D3D3D3"
                              />
                              <circle
                                        className="DC2"
                                        cx="801.3"
                                        cy="251.8"
                                        data-name="DC"
                                        fill="#D3D3D3"
                                        onClick={undefined}
                                        opacity="1"
                                        r="5"
                                        stroke="#FFFFFF"
                                        strokeWidth="1.5"
                              />
                    </g>
          </g>
</svg>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <title>
              Yo Man! Wadd up......
</title>,
            <g
              className="outlines"
>
              <g
                            className="DC state"
              >
                            <path
                                          className="DC1"
                                          d="M801.8,253.8 l-1.1-1.6 -1-0.8 1.1-1.6 2.2,1.5z"
                                          fill="#D3D3D3"
                            />
                            <circle
                                          className="DC2"
                                          cx="801.3"
                                          cy="251.8"
                                          data-name="DC"
                                          fill="#D3D3D3"
                                          onClick={undefined}
                                          opacity="1"
                                          r="5"
                                          stroke="#FFFFFF"
                                          strokeWidth="1.5"
                            />
              </g>
</g>,
          ],
          "className": "us-state-map",
          "height": 593,
          "viewBox": "0 0 959 593",
          "width": 959,
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Yo Man! Wadd up......",
            },
            "ref": null,
            "rendered": "Yo Man! Wadd up......",
            "type": "title",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [],
                <g
                  className="DC state"
>
                  <path
                                    className="DC1"
                                    d="M801.8,253.8 l-1.1-1.6 -1-0.8 1.1-1.6 2.2,1.5z"
                                    fill="#D3D3D3"
                  />
                  <circle
                                    className="DC2"
                                    cx="801.3"
                                    cy="251.8"
                                    data-name="DC"
                                    fill="#D3D3D3"
                                    onClick={undefined}
                                    opacity="1"
                                    r="5"
                                    stroke="#FFFFFF"
                                    strokeWidth="1.5"
                  />
</g>,
              ],
              "className": "outlines",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <path
                      className="DC1"
                      d="M801.8,253.8 l-1.1-1.6 -1-0.8 1.1-1.6 2.2,1.5z"
                      fill="#D3D3D3"
/>,
                    <circle
                      className="DC2"
                      cx="801.3"
                      cy="251.8"
                      data-name="DC"
                      fill="#D3D3D3"
                      onClick={undefined}
                      opacity="1"
                      r="5"
                      stroke="#FFFFFF"
                      strokeWidth="1.5"
/>,
                  ],
                  "className": "DC state",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "DC1",
                      "d": "M801.8,253.8 l-1.1-1.6 -1-0.8 1.1-1.6 2.2,1.5z",
                      "fill": "#D3D3D3",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "path",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "DC2",
                      "cx": "801.3",
                      "cy": "251.8",
                      "data-name": "DC",
                      "fill": "#D3D3D3",
                      "onClick": undefined,
                      "opacity": "1",
                      "r": "5",
                      "stroke": "#FFFFFF",
                      "strokeWidth": "1.5",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "circle",
                  },
                ],
                "type": "g",
              },
            ],
            "type": "g",
          },
        ],
        "type": "svg",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
